src/app/memory-monitor.ts

```diff
@@ -0,0 +1,88 @@
+import * as electron from 'electron';
+
+import { logger } from '../common/logger';
+import { config } from './config-handler';
+import { windowHandler } from './window-handler';
+import { windowExists } from './window-utils';
+
+class MemoryMonitor {
+    private memoryInfo: Electron.ProcessMemoryInfo | undefined = undefined;
+    private isInMeeting: boolean;
+    private canReload: boolean;
+
+    private readonly maxIdleTime: number;
+    private readonly memoryThreshold: number;
+    private readonly memoryRefreshThreshold: number;
+
+    constructor() {
+        this.isInMeeting = false;
+        this.canReload = true;
+        this.maxIdleTime = 4 * 60 * 60 * 1000; // 4 hours
+        this.memoryThreshold = 800 * 1024; // 800MB
+        this.memoryRefreshThreshold = 60 * 60 * 1000; // 1 hour
+    }
+
+    /**
+     * Sets process memory from ipc events every hour
+     * and refreshes the client if the conditions passes
+     *
+     * @param memoryInfo {Electron.ProcessMemoryInfo}
+     */
+    public setMemoryInfo(memoryInfo: Electron.ProcessMemoryInfo): void {
+        this.memoryInfo = memoryInfo;
+        this.validateMemory();
+    }
+
+    /**
+     * Sets the web app's RTC meeting status
+     *
+     * @param isInMeeting {boolean} whether user is in an active RTC meeting
+     */
+    public setMeetingStatus(isInMeeting: boolean): void {
+        this.isInMeeting = isInMeeting;
+    }
+
+    /**
+     * Validates the predefined conditions and refreshes the client
+     */
+    private validateMemory(): void {
+        (electron.powerMonitor as any).querySystemIdleTime((time) => {
+            const idleTime = time * 1000;
+            if (!this.isInMeeting
+                && windowHandler.isOnline
+                && this.canReload
+                && idleTime > this.maxIdleTime
+                && (this.memoryInfo && this.memoryInfo.private > this.memoryThreshold)
+            ) {
+                const { memoryRefresh } = config.getConfigFields([ 'memoryRefresh' ]);
+                if (memoryRefresh) {
```

Comment:
same as above, please move this check (inverted) before the above `if` condition.